### ISSUE TITLE
Windows: opening SC via system registered document types faulty

### ISSUE THREAD

**Issue by bagong** [supercollider/supercollider#2022]
Reproducer for two different of possibly related issues:

First
- start SC by clicking scd or sc document (works)
- boot server: Fails!

Second
- start SC from start menu or by clicking executable (works)
- now try to load another document into running SC by (double)-clicking the document: Fails.

Thanks PyoungRyang Ko for reporting

**Comment by llloret** [comment]
Do you know where is the installer generated?

**Comment by bagong** [comment]
yes, it's an NSI script in platform/Windows

**Comment by llloret** [comment]
First issue can be solved by adding the App Path information to the registry: https://msdn.microsoft.com/en-us/library/windows/desktop/ee872121(v=vs.85).aspx. It works adding it manually. Let's see how to put that into the nsis script, but I am not familiar with nsis.

**Comment by bagong** [comment]
That sounds possible, there are variables for that in NSIS, they are used to install SC as well. I'll have to freshen up my memory. I am sure you could easily learn by reading the script, but I can also try, if you prefer... Or do it on Gitter "together"?

**Comment by llloret** [comment]
What is the expected behaviour in case 2? Should it open it in a new instance, or on a running instance?

**Comment by bagong** [comment]
I think a new tab in the running instance would be the expected behaviour. I don't even know if you can have two instances ;) I'll try on the Mac... Yep, it opens in a new tab.

**Comment by llloret** [comment]
I don't have nsis, but I think we would need to add something like this:
```
; Add path information
  WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\App Paths\scide.exe" "" "$INSTDIR\scide.exe"
  WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\App Paths\scide.exe" "Path" "$INSTDIR"  
```

Just over this:
```
; Create Startmenu item
  SetShellVarContext all
```

As I say, that should fix issue 1. Regarding issue 2, it probably needs some handling within scide.

**Comment by llloret** [comment]
We can see in this gitter thing if you want

**Comment by bagong** [comment]
Okay, I am bagong ;)

I was just writing:

Okay, I will try this with NSIS.
I also just saw there is an FileAssociation plugin that can be included for NSIS. If I don't get there with your suggestion, I'll try that.

Is to 2, there is also som build configuration option in VS relating to Unicode vs other code page system, there is also something like "Culture", which is set to US-American atm... There is something VS specific in this, the other build system don't have the problem as sharply as VS

**Comment by llloret** [comment]
Issue 1 is fixed with PR #2028.

**Comment by bagong** [comment]
Issue 2 is fixed by PR #2029. ¡Viva Luis!
